### PR TITLE
Add weighted average aggregation

### LINKED ISSUE #19: FedAvg weighted average implementation
The current FedAvg implementation doesn't support weighted average aggregation. We need to support this. This issue possibly is dependent on #18.

https://github.com/bnnthang/RealFL/blob/c3510b7dfc32b54e74eda4207fc335fc3c82d837/Server/src/main/java/com/bnnthang/fltestbed/Server/AggregationStrategies/FedAvg.java#L12-L24

### PR DESCRIPTION
**Pull request by ttanv**
This PR is to close #19.

### CODE REVIEW

File: CommonUtils/src/main/java/com/bnnthang/fltestbed/commonutils/models/TrainingConfiguration.java [hunk 1]
```diff
@@ -68,14 +68,12 @@ public class TrainingConfiguration {
     private ArrayList<ArrayList<Double>> distributionRatiosByLabels;
 
     /**
-     * If clients should be dropped
+     * Ture if clients should be dropped
```
Comment (thangnbn) [inline CommonUtils/src/main/java/com/bnnthang/fltestbed/commonutils/models/TrainingConfiguration.java]: @tem556 Spelling error.

File: Server/config.json [hunk 2]
```diff
@@ -5,5 +5,8 @@
 				   [0.25, 0.5, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25],
 				   [0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25]],
     "useDropping": true,
-    "dropping": [{"round":1,"#clients":1},{"round":2,"#clients":1} ]
+    "dropping": [{"round":1,"#clients":1},{"round":2,"#clients":1} ],
+
+    "useWeights": true
```
Comment (thangnbn) [inline Server/config.json]: @tem556 This is not what we intended. We want the weights to be the same as the data partition ratios. For example, if we divide the dataset 20% : 20% : 60% among the clients then the weights would be 0.2 : 0.2 : 0.6.

Comment (ttanv) [reply Server/config.json]: I see, I think I misunderstood

**Comment by ttanv** [comment]
@bnnthang I think this pull request can be merged now?
Also, please check out the attempt at using factory pattern for repository. (Unrelated to FedAvg issue)

**Pull request merged**